cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,303 @@
+		fmt.Fprintln(w, x.tabline(&sg, extraWidth))
+	}
+
+	return nil
+}
+
+type saveCmd struct {
+	waitMixin
+	timeOpts
+	Homes      []string `long:"homes"`
+	Positional struct {
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes"`
+}
+
+var shortSaveHelp = i18n.G("Save a snapshot of the current data")
+var longSaveHelp = i18n.G(`
+The save command creates a snapshot of the current data for the given snaps.
+`)
+
+func (x *saveCmd) Execute([]string) error {
+	cli := Client()
+	changeID, err := cli.SnapshotMany(installedSnapNames(x.Positional.Snaps), x.Homes)
+	if err != nil {
+		return err
+	}
+	chg, err := x.wait(cli, changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	var shID snapshotID
+	chg.Get("snapshot-id", &shID)
+	y := &savedCmd{
+		timeOpts: x.timeOpts,
+		ID:       shID,
+	}
+	y.Positional.Snaps = x.Positional.Snaps
+	return y.Execute(nil)
+}
+
+type loseCmd struct {
+	waitMixin
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortLoseHelp = i18n.G("Delete a snapshot")
+var longLoseHelp = i18n.G(`
+The lose command deletes a snapshot.
+`)
+
+func (x *loseCmd) Execute([]string) error {
+	cli := Client()
+	snaps := installedSnapNames(x.Positional.Snaps)
+	changeID, err := cli.LoseSnapshot(uint64(x.Positional.ID), snaps)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(cli, changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d of snaps %s is lost to the ages.\n"), x.Positional.ID, strutil.Quoted(snaps))
+	} else {
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d is lost to the ages.\n"), x.Positional.ID)
+	}
+	return nil
+}
+
+type checkCmd struct {
+	waitMixin
+	Homes      string `long:"homes"`
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortCheckHelp = i18n.G("Check a snapshot")
+var longCheckHelp = i18n.G(`
+The check command checks a snapshot against its hashsums.
+`)
+
+func (x *checkCmd) Execute([]string) error {
+	cli := Client()
+	snaps := installedSnapNames(x.Positional.Snaps)
+	homes := strings.Split(x.Homes, ",")
+	changeID, err := cli.CheckSnapshot(uint64(x.Positional.ID), snaps, homes)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(cli, changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// TODO: also mention the home archives that were actually checked
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d is probably *fine*, at least for snaps %s.\n"),
```

Comment:
This might also use a more straightforward "Snapshot %d looks okay.". The "at least" here also puts some doubt that seems strange in this context. Feels like we might have actual hints that something else might be broken.